up-transport-vsomeip/src/message_conversions.rs

```diff
@@ -233,32 +233,22 @@ where {
             client_id,
             session_id
         );
-        let ok = {
+
+        let (commstatus, vsomeip_msg_type) = {
             if let Some(commstatus) = umsg.attributes.commstatus {
-                let commstatus = commstatus.enum_value_or(UCode::UNIMPLEMENTED);
-                trace!("UMessage Response commstatus was set to: {commstatus:?}");
-                commstatus == UCode::OK
+                (commstatus.enum_value_or(UCode::UNIMPLEMENTED),message_type_e::MT_ERROR)
+                
             } else {
-                trace!("UMessage Response had no commstatus set");
-                true
+                (UCode::UNIMPLEMENTED, message_type_e::MT_RESPONSE)
             }
         };
-        if ok {
-            vsomeip_msg
-                .get_message_base_pinned()
-                .set_return_code(vsomeip::return_code_e::E_OK);
-            vsomeip_msg
-                .get_message_base_pinned()
-                .set_message_type(message_type_e::MT_RESPONSE);
-        } else {
-            // TODO: Perform mapping from uProtocol UCode contained in commstatus into vsomeip::return_code_e
-            vsomeip_msg
-                .get_message_base_pinned()
-                .set_return_code(vsomeip::return_code_e::E_NOT_OK);
-            vsomeip_msg
-                .get_message_base_pinned()
-                .set_message_type(message_type_e::MT_ERROR);
-        }
+
+        vsomeip_msg
+            .get_message_base_pinned()
+            .set_return_code(Self::ucode_to_return_code_e(commstatus));
```

Comment:
This handles UMessage commstatus -> vsomeip message status mapping.

We also need to handle vsomeip message status -> UMessage commstatus mapping over [here](https://github.com/eclipse-uprotocol/up-transport-vsomeip-rust/blob/acbb0d0c9b8b48dd35c74f461e97151f1e922000/up-transport-vsomeip/src/message_conversions.rs#L507) as mentioned in the task following the same [spec](https://github.com/eclipse-uprotocol/up-transport-vsomeip-rust/pull/22).

Reply:
> We also need to handle vsomeip message status -> UMessage ...
my mistake. Brain skipped over